### PR TITLE
Unit tests for hmi language handler

### PR DESCRIPTION
**Pull request by VVeremjova**
Tests were moved from PASA

Related to [APPLINK-22720](https://adc.luxoft.com/jira/browse/APPLINK-22720)

Please review @OHerasym, @okozlovlux, @LevchenkoS, @VProdanov, @dtrunov, @nk0leg

### CODE REVIEW

**Comment by LevchenkoS** [comment]
@VVeremjova, reviewed.

File: src/components/application_manager/test/hmi_language_handler_test.cc [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdint.h>
+
+#include "gtest/gtest.h"
+#include "application_manager/application_manager.h"
+#include "application_manager/hmi_language_handler.h"
+#include "application_manager/state_controller.h"
+#include "application_manager/mock_application_manager.h"
```
Comment (okozlovlux) [inline src/components/application_manager/test/hmi_language_handler_test.cc]: @VVeremjova , do you need to include both **application_manager** and **mock_application_manager** in this test?

File: src/components/application_manager/test/hmi_language_handler_test.cc [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+  hmi_language_handler_->on_event(ev3);
+}
+
+TEST_F(HmiLanguageHandlerTest, OnEvent_AllReceivedLanguagesMismatch_SUCCESS) {
+  Event ev1(hmi_apis::FunctionID::UI_GetLanguage);
+  hmi_language_handler_->on_event(ev1);
+  Event ev2(hmi_apis::FunctionID::VR_GetLanguage);
+  hmi_language_handler_->on_event(ev2);
+
+  const Apps& apps = hmi_language_handler_->get_apps();
```
Comment (okozlovlux) [inline src/components/application_manager/test/hmi_language_handler_test.cc]: @VVeremjova **kApps** ?

File: src/components/application_manager/test/hmi_language_handler_test.cc [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+      hmi_apis::Common_Language::eType::EN_US,
+      hmi_apis::Common_Language::eType::EN_US,
+      hmi_apis::Common_Language::eType::EN_US);
+
+  InitHMIAtciveLanguage(hmi_apis::Common_Language::eType::RU_RU,
+                        hmi_apis::Common_Language::eType::RU_RU,
+                        hmi_apis::Common_Language::eType::RU_RU);
+
+  ApplicationSet app_set;
+  const uint32_t app_count = 5u;
```
Comment (okozlovlux) [inline src/components/application_manager/test/hmi_language_handler_test.cc]: @VVeremjova **kAppCount**

**Comment by okozlovlux** [comment]
@VVeremjova, reviewed. 
But, please, check const variable's names.

File: src/components/application_manager/test/hmi_language_handler_test.cc (hunk 3 above)
Comment (VVeremjova) [inline src/components/application_manager/test/hmi_language_handler_test.cc]: @okozlovlux See 9e4cec1

File: src/components/application_manager/test/hmi_language_handler_test.cc (hunk 2 above)
Comment (VVeremjova) [inline src/components/application_manager/test/hmi_language_handler_test.cc]: @okozlovlux See 9e4cec1

File: src/components/application_manager/test/hmi_language_handler_test.cc (hunk 1 above)
Comment (VVeremjova) [inline src/components/application_manager/test/hmi_language_handler_test.cc]: @okozlovlux Yes, both of them are needed

**Comment by VProdanov** [comment]
@VVeremjova Reviewed

**Comment by OHerasym** [comment]
Reviewed

**Comment by nk0leg** [comment]
@VVeremjova, reviewed.

**Pull request merged**